### PR TITLE
Quick Sort 3 way in Ruby

### LINKED ISSUE #2170: Quick Sort 3 way
- [ ] C - @Moddy2024
- [x] C++
- [ ] C# - @Harikrishnan6336
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [ ] Java - @IPSITA2000
- [ ] JS - @atrip10
- [ ] Kotlin
- [ ] Php - @BellatrixLestrangee
- [x] Python
- [ ] Readme - @SomyaGandhi
- [x] Ruby - @HariniJeyaraman
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by HariniJeyaraman**
Fixes #2170

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Implemented QuickSort in Ruby
- 

#### Languages Used:
-  Ruby
- 

#### Files Added:
-  quick_sort_3_way.rb
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
please remove the unnecessary files

**Comment by HariniJeyaraman** [comment]
@asha15 Done. Removed the extra file

File: Quicksort_3_way/quick_sort_3_way.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#3 way Quick Sort in Ruby
```
Comment (asha15) [inline Quicksort_3_way/quick_sort_3_way.rb:1]: keep a line space after this

File: Quicksort_3_way/quick_sort_3_way.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+def quick_sort(a,lo,hi)
```
Comment (asha15) [inline Quicksort_3_way/quick_sort_3_way.rb:2]: keep a space after comma

File: Quicksort_3_way/quick_sort_3_way.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        elsif a[i] > pivot
+            temp = a[i]
+            a[i] = a[gt]
+            a[gt] = temp
+            gt -= 1
+        else
+            i += 1
+        end
+    end
+    return lt,gt
```
Comment (asha15) [inline Quicksort_3_way/quick_sort_3_way.rb:34]: keep space after comma

File: Quicksort_3_way/quick_sort_3_way.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+puts "Enter the size of array : "
+n = gets
+puts "Enter the values for the array : "
+$i = 0
+$num = Integer(n)
+arr = Array.new
+while $i < $num  do
+    arr[$i] = gets
+    $i +=1
+end
```
Comment (asha15) [inline Quicksort_3_way/quick_sort_3_way.rb:46]: keep a line space before and after the while loop

File: Quicksort_3_way/quick_sort_3_way.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+12
+13
+
+After sorting : 
+10
+12
+13
+14
+=end
+
```
Comment (asha15) [inline Quicksort_3_way/quick_sort_3_way.rb:68]: remove this line

**Comment by HariniJeyaraman** [comment]
@asha15 I have made the changes. Please review again

**Comment by asha15** [comment]
Good job!
please squash the commits , we are good to merge

**Comment by HariniJeyaraman** [comment]
@asha15 Sqaushed commits.

**Comment by HariniJeyaraman** [comment]
@asha15 Please review again. I have squashed the commits

File: Quicksort_3_way/quick_sort_3_way.rb (hunk 5 above)
Comment (HariniJeyaraman) [reply Quicksort_3_way/quick_sort_3_way.rb]: done ! Please review my code

**Pull request merged**